src/Exception/UnencodableValueException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace MongoDB\Exception;
+
+use InvalidArgumentException;
+
+use function get_debug_type;
+use function sprintf;
+
+class UnencodableValueException extends InvalidArgumentException implements Exception
```

Comment:
Some thoughts on this.

For GridFS, we lumped its three exceptions under a `MongoDB\GridFS\Exception` namespace. Is that something we'd want to do for Codec as well, assuming these exceptions will only be used by classes in `MongoDB\Codec`?

I think the only trade-off is that it makes the `use` statement a bit longer, but it does convey that these are codec exceptions.

And a separate question for @GromNaN, since you suggested using dedicated exceptions. Do you feel strongly about having a separate exception for each method, or could there be a common UnsupportedValueException thrown by both `encode()` and `decode()`? I can't predict all ways the codec feature will be used, but I don't expect it'd be common for applications to encode _and_ decode within the same `try/catch` scope.

Reply:
In hindsight, I regret that we introduced so many exceptions in the initial release of PHPC. There are cases where different exceptions made sense (e.g. CommandException, ServerException, WriteException), but there are an equal number of mistakes (e.g. SSLConnectionException).

In this case, both exceptions expose a `$value` property and there should be little confusion about the context. I'd very much be in favor of using a common exception to reduce the cognitive and documentation overhead.